sources/packages/api/src/route-controllers/student/student.controller.ts

```diff
@@ -308,4 +313,48 @@ export class StudentController extends BaseController {
 
     stream.pipe(response);
   }
+
+  @Get("application-summary")
+  async getStudentApplicationSummary(
+    @UserToken() userToken: IUserToken,
+  ): Promise<StudentApplicationDTO[]> {
+    const existingStudent = await this.studentService.getStudentByUserName(
+      userToken.userName,
+    );
+    if (!existingStudent) {
```

Comment:
I believe that we can add the student id to and "Student Token" and move the validation if the student is present to a decorator. We can create a tech debit for it. The idea would be to try to keep the controller clean.

Reply:
i agree